apps/wallet-dashboard/hooks/useGetStardustIndexerOutputs.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+import { useGetCurrentEpochStartTimestamp } from '@/hooks';
+import {
+    mapStardustBasicOutputs,
+    mapStardustNftOutputs,
+    PageParams,
+    StardustIndexerOutput,
+    TimeUnit,
+    useStardustIndexerClientContext,
+} from '@iota/core';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+
+const LIMIT_PER_REQ = 50;
+
+export function useGetStardustIndexerOutputs(address: string) {
+    const { data: currentEpochMs } = useGetCurrentEpochStartTimestamp();
+    const { stardustIndexerClient } = useStardustIndexerClientContext();
+
+    const fetchPaginatedOutputs = async (
+        mapFn: (output: StardustIndexerOutput) => IotaObjectData,
+        fetchFn: (address: string, params: PageParams) => Promise<StardustIndexerOutput[]>,
+    ) => {
+        const allOutputs = [];
+        let page = 1;
+        let hasMoreData = true;
+
+        try {
+            while (hasMoreData) {
+                const outputs = await fetchFn(address, { page, pageSize: LIMIT_PER_REQ });
+
+                if (!outputs || outputs.length === 0) {
+                    hasMoreData = false;
+                } else {
+                    allOutputs.push(...outputs);
+                    page++;
+                }
+            }
+        } catch (e) {
+            console.error(e);
+        }
+
+        return allOutputs.map(mapFn);
+    };
```

Comment:
we could follow a similar code structure we have in similar hooks

```suggestion
    const fetchPaginatedStardustSharedObjects = async (
        mapFn: (output: StardustIndexerOutput) => IotaObjectData,
        fetchFn: (
            address: string,
            params: { page: number; limit: number },
        ) => Promise<StardustIndexerOutput[]>,
    ) => {
        let page = 1;
        const allData: StardustIndexerOutput[] = [];
        try {
            do {
                const data = await fetchFn(address, { page, limit: LIMIT_PER_REQ });
                if (!data || !data.length) {
                    break;
                }
                allData.push(...data);
                page++;
            } while (page);
        } catch (e) {
            console.error(e);
        }

        return allData.map(mapFn);
    };
```